test/user_test.exs

```diff
@@ -0,0 +1,41 @@
+defmodule DoAuth.UserTest do
+  use ExUnit.Case
+
+  alias Uptight.Result
+  alias Uptight.Text, as: T
+  alias DoAuth.User
+
+  describe "reserve_identity/3" do
+    setup do
+      %Result.Ok{ok: email} = T.new("user@mail.com")
+      %Result.Ok{ok: nickname} = T.new("user")
+
+      on_exit(fn -> File.rm_rf!(Path.join(["db", "nonode@nohost"])) end)
+
+      %{email: email, nickname: nickname}
+    end
+
+    test "with unused email allows to reserve identity", %{email: %T{text: email_text} = email, nickname: nickname} do
+      %Result.Ok{ok: pid} = User.reserve_identity(email, nickname)
+
+      assert is_pid(pid)
+
+      assert %{
+        email: ^email,
+        nickname: ^nickname,
+        cred: %Uptight.Base.Urlsafe{},
```

Comment:
The `cred` currently isn't `%Uptight.Base.Urlsafe{}` struct, but a plain map. Is it intended of not?

Reply:
It's not indended, I think that we should point to credential store rather than copying it into the state process.